scala/build/maintenance.Dockerfile

```
@@ -116,22 +116,33 @@
 
 # Install Python and required packages
 # Optional: If you need venv for virtual environments
-RUN apt-get update && apt-get install -y software-properties-common && \
+# Install required packages and Python 3.12
+RUN apt-get update && apt-get install -y software-properties-common curl && \
```

Hadolint

[info] Avoid additional packages by specifying `--no-install-recommends`

(DL3015)